pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -102,10 +102,31 @@ func RunStartBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args
 		return cmdutil.UsageError(cmd, "Must pass a name of a BuildConfig or specify build name with '--from-build' flag")
 	}
 
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
 	name := buildName
 	isBuild := true
-	if len(name) == 0 {
-		name = args[0]
+	if len(name) == 0 && len(args[0]) > 0 {
+		parts := strings.Split(args[0], "/")
```

Comment:
I am not using the resource builder here because it would try to get the buildConfig (which we are not interested in) in order to start the build. Instead, I think using the RESTMapper to resolve the resource type is enough.